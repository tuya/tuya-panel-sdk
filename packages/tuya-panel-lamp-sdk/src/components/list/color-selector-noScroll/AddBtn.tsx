import React from 'react';
import { ViewStyle, StyleProp } from 'react-native';
import { IconFont } from 'tuya-panel-kit';
import Button from './Button';

export interface IProps {
  btnColor: string;
  iconColor?: string;
  style?: StyleProp<ViewStyle>;
  onPress?: () => void;
}

const AddBtn: React.FunctionComponent<IProps> = ({
  style,
  iconColor,
  btnColor,
  onPress,
}: IProps) => (
  <Button color={btnColor} style={style} onPress={onPress}>
    <IconFont name="plus" size={18} color={iconColor} />
  </Button>
);

export default AddBtn;
