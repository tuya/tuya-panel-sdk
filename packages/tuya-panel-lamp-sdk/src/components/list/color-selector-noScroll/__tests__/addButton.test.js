import React from 'react';
import { shallow } from 'enzyme';
import AddBtn from '../AddBtn';

describe('AddBtn components', () => {
  it('basic render', () => {
    const wrapper = shallow(<AddBtn btnColor={'#333'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
