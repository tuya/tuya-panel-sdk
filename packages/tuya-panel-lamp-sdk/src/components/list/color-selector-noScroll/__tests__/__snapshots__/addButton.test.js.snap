// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBtn components basic render 1`] = `
<Button
  color="#333"
>
  <Component
    name="plus"
    size={18}
  />
</Button>
`;
